animation_nodes/nodes/action/chain_action_list.pyx

```diff
@@ -0,0 +1,131 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode
+from ... graphics import Rectangle
+from ... graphics.drawing_2d import drawVerticalLine
+from ... data_structures cimport (
+    BoundedAction, BoundedActionEvaluator,
+    FloatList, BooleanList
+)
+
+class ChainActionListNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_ChainActionListNode"
+    bl_label = "Chain Action List"
+    errorHandlingType = "EXCEPTION"
+
+    __annotations__ = {}
+    __annotations__["relative"] = BoolProperty(name = "Relative", default = True,
+        description = "Use last frame of first action as starting point of the second",
+        update = propertyChanged)
+
+    def create(self):
+        self.newInput("Action List", "Action list", "list")
+        self.newOutput("Action", "Chained Action")
+
+    def draw(self, layout):
+        layout.prop(self, "relative")
+
+    def execute(self, list):
+        if list is None:
+            return list
+
+        length = len(list)
+
+        if length == 0:
+            return None
+
+        if length < 2:
+            return list[0]
+        
+        r = list[0]
+
+        if not isinstance(r, BoundedAction):
+            self.raiseErrorMessage("only bounded actions can be chained")
+
+        i = 1
+        while i < length:
+            b = list[i]
+            if isinstance(b, BoundedAction):
+                r = ChainAction(r, b, self.relative)
+            i += 1
+
+        return r
+
+cdef class ChainAction(BoundedAction):
```

Comment:
Aren't those classes identical to those in the Chain Actions node? Can't we just use the already defined classes?